File: Dsa-day123.cpp
class Solution {
public:
    bool kLengthApart(vector<int>& nums, int k) {
        int prev = -1 - k;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] == 1) {
                if (i - prev <= k) return false;
                prev = i;
            }
        }
        return true;
    }
};

File: Dsa-day123.py
class Solution:
    def kLengthApart(self, nums: List[int], k: int) -> bool:
        prev = -1 - k
        for i in range(len(nums)):
            if nums[i] == 1:
                if i - prev <= k:
                    return False
                prev = i
        return True